### ISSUE TITLE
WebSocket instance with no server-side message subscriber appears to close itself

### ISSUE THREAD

**Issue by jeblackburn** [vtortola/WebSocketListener#106]
I have duplicated much of the EchoServer code in a project of my own, and have found behavior that I thought I'd ask about.

If I receive a connection request and call `await webSocket.ReadStringAsync` in a while loop, everything works fine. If I don't make that looping call, but just keep a reference to the webSocket around, occasionally sending messages to the client, the socket closes itself within a few seconds. 

It will live indefinitely if I set `PingTimeout = Timeout.Infinite` when constructing the WebSocketListener.

My question is this: Is it a known requirement that the server must subscribe to messages from the client in order for the connection to remain alive and healthy?

I've attached the server-side code and a C# client I'm using to test.

[Server.cs.txt](https://github.com/vtortola/WebSocketListener/files/956427/Server.cs.txt)
[Program.client.cs.txt](https://github.com/vtortola/WebSocketListener/files/956431/Program.client.cs.txt)
[Program.server.cs.txt](https://github.com/vtortola/WebSocketListener/files/956433/Program.server.cs.txt)

Thanks for the help/info.

Jon

**Comment by vtortola** [comment]
> My question is this: Is it a known requirement that the server must subscribe to messages from the client in order for the connection to remain alive and healthy?

Yes it is. Ping frames are interleaved  with data frames. When you "read" you are pulling data from the socket, and the component is removing the control frames so you get only the data. The Microsoft client behaves in a similar way as far as I remember, it cannot answer the pings if you are not continuously reading from it. 

Now that I think about it, may be it should be indicated somewhere in the documentation :P
